src/consumer/stream_consumer.rs

```diff
@@ -49,76 +48,86 @@ unsafe fn disable_nonempty_callback(queue: &NativeQueue) {
     rdsys::rd_kafka_queue_cb_event_enable(queue.ptr(), None, ptr::null_mut())
 }
 
-struct WakerSlab {
-    wakers: Mutex<Slab<Option<Waker>>>,
+/// Data structure to store waker instances from idle streams.
+/// This queue wakes up the wakers in FIFO order.
+/// On registration streams receive an id they can reuse for setting and removing wakers.
+///
+/// Implementation is optimized for adding new wakers and waking all of them.
+/// Removing a waker is less efficient.
```

Comment:
question: Wouldn't a `std::collections::VecDeque` be a better fit here?

Reply:
Given that removing a waker requires an identifer (e.g. the UUID returned when registering as an arg), it's still not optimized for removing with a `VecDeque`. And since `VecDeque` is a ring buffer, adding to it is less optimized when full than with a `Vec`.

Reply:
Could also use a btreemap. 😆 
But I think that everything else than waking up all wakers is not in the critical and therefore I wouldn't optimize for that.

Reply:
Agreed.